mlflow/server/js/src/components/ExperimentViewUtil.js

```diff
@@ -402,10 +405,24 @@ export default class ExperimentViewUtil {
     return mergedRows;
   }
 
+  /**
+   * Given an array of row metadata returned by getRowRenderMetadata, returns an array of run IDs
+   * sorted by display order.
+   */
+  static getRunIdsSortedByDisplayOrder(rowMetadatas) {
+    return rowMetadatas.flatMap(rowMetadata => {
+      if (!rowMetadata.isParent) {
+        return [];
+      }
+      return _.concat([rowMetadata.runId], rowMetadata.childrenIds || []);
+    });
+  }
+
   static getRows({ runInfos, sortState, paramsList, metricsList, tagsList, runsExpanded, getRow }) {
     const mergedRows = ExperimentViewUtil.getRowRenderMetadata(
       { runInfos, sortState, paramsList, metricsList, tagsList, runsExpanded });
-    return mergedRows.map((rowMetadata) => getRow(rowMetadata));
+    const sortedRunIds = mergedRows.map(mergedRow => mergedRow.runId);
+    return mergedRows.map((rowMetadata) => getRow({...rowMetadata, sortedRunIds}));
```

Comment:
I guess `displayIndex` was already there?

Reply:
Ah `displayIndex` is there now (it's a key in `rowMetadata`)  since it's returned by `getRowRenderMetadata`